crates/primitives/src/precompile.rs

```diff
@@ -2,11 +2,30 @@ use crate::{Bytes, Env};
 use core::fmt;
 use dyn_clone::DynClone;
 use std::{boxed::Box, string::String, sync::Arc};
+use revm_precompile::PrecompileOutput;
 
 /// A precompile operation result.
 ///
 /// Returns either `Ok((gas_used, return_bytes))` or `Err(error)`.
-pub type PrecompileResult = Result<(u64, Bytes), PrecompileError>;
+pub enum PrecompileErrors {
+    Error(PrecompileError),
```

Comment:
You could implement `impl From<PrecompileError> for PrecompileErrors {`

And transform from previous `return Err(Error::OutOfGas);` to return Err(Error::OutOfGas.into());